src/http.rs

```diff
@@ -184,6 +250,36 @@ impl HttpClient {
     }
 }
 
+fn build_timed_url(
+    url: String,
+    endpoint: &str,
+    transaction_time: Option<DateTime<FixedOffset>>,
+    valid_time: Option<DateTime<FixedOffset>>,
+) -> String {
+    match (transaction_time, valid_time) {
+        (None, None) => format!("{}/{}", url, endpoint),
+        (Some(tx), None) => format!(
+            "{}/{}?transaction-time={}",
+            url,
+            endpoint,
+            tx.format("%Y-%m-%dT%H:%M:%S%Z").to_string()
```

Comment:
What about putting this format string (`"%Y-%m-%dT%H:%M:%S%Z"`) into a constant, so we can change it easly later :slightly_smiling_face: